presto-main/src/main/java/com/facebook/presto/sql/planner/plan/StatisticAggregations.java

```diff
@@ -25,31 +25,51 @@
 import com.google.common.collect.ImmutableList;
 import com.google.common.collect.ImmutableMap;
 
+import javax.annotation.concurrent.Immutable;
+
 import java.util.List;
 import java.util.Map;
 import java.util.Optional;
+import java.util.stream.Collectors;
 
 import static com.google.common.base.Preconditions.checkArgument;
 import static java.util.Objects.requireNonNull;
 
 public class StatisticAggregations
 {
+    // Added to preserve the order for serializing and deserializing
+    // map in c++ worker doesn't represent insertion order
+    private final List<VariableAggregation> aggregationList;
     private final Map<VariableReferenceExpression, Aggregation> aggregations;
     private final List<VariableReferenceExpression> groupingVariables;
 
     @JsonCreator
     public StatisticAggregations(
-            @JsonProperty("aggregations") Map<VariableReferenceExpression, Aggregation> aggregations,
+            @JsonProperty("aggregationList") List<VariableAggregation> aggregationList,
             @JsonProperty("groupingVariables") List<VariableReferenceExpression> groupingVariables)
+    {
+        this.aggregationList = ImmutableList.copyOf(requireNonNull(aggregationList, "aggregationList is null"));
+        this.groupingVariables = ImmutableList.copyOf(requireNonNull(groupingVariables, "groupingVariables is null"));
+        ImmutableMap.Builder<VariableReferenceExpression, Aggregation> aggregationMapBuilder = ImmutableMap.builder();
+        for (VariableAggregation aggregation : aggregationList) {
+            aggregationMapBuilder.put(aggregation.variableReferenceExpression, aggregation.aggregation);
+        }
+        aggregations = aggregationMapBuilder.build();
+    }
+
+    public StatisticAggregations(
```

Comment:
nit: Is this still required? Thanks!

Reply:
Yes, basically in java code base, they're still heavily rely on the map structure, so if we only keep the List, the lookup would not be efficient.

So essentially we only introduced the List structure for serialization and deserialization, internally the logic remained the same.